src/content_scripts/autocorrect.js

```diff
@@ -0,0 +1,400 @@
+
+    if (!output) {
+        strm += anumber;
+    }
+
+    return strm;
+}
+
+/**
+ * Get first difference index.
+ *
+ * @param {string} a
+ * @param {string} b
+ * @returns {number}
+ */
+function firstDifferenceIndex(a, b) {
+    if (a === b) {
+        return -1;
+    }
+    let i = 0;
+    while (a[i] === b[i]) {
+        ++i;
+    }
+    return i;
+}
+
+/**
+ * Autocorrect on text input even by evaluating the keys and replacing the characters/string.
+ *
+ * @param {Object} event
+ * @returns {void}
+ */
+function autocorrect(event) {
+    // console.log('keydown', event.key, event.key.length, event.keyCode);
+    if (!((event.key.length === 0 || event.key.length === 1 || event.keyCode === 13 || event.key === "Unidentified") && !event.ctrlKey && !event.metaKey && !event.altKey)) {
+        return;
+    }
+    if (!symbolpatterns) {
+        throw new Error("Emoji autocorrect settings have not been received. Do not autocorrect.");
+    }
+    const target = event.target;
+    const caretposition = getCaretPosition(target);
+    if (caretposition) {
+        const value = target.value || target.innerText;
+        let deletecount = 0;
+        let insert = value.slice(caretposition - 1, caretposition); // event.key;
+        let output = false;
+        // Use Unicode smart quotes
+        if (quotes && (insert === "'" || insert === '"')) {
+            const prevouschar = value.slice(caretposition < 2 ? 0 : caretposition - 2, caretposition - 1);
+            // White space
+            const re = /^\s*$/;
+            if (insert === "'") {
+                insert = re.test(prevouschar) ? "‘" : "’";
+            } else if (insert === '"') {
+                insert = re.test(prevouschar) ? "“" : "”";
+            }
+            deletecount = 1;
+            output = true;
+        }
+        const prevoustext = value.slice(caretposition < (longest + 1) ? 0 : caretposition - (longest + 1), caretposition - 1);
+        const regexResult = symbolpatterns.exec(prevoustext);
+        // Autocorrect Unicode Symbols
+        if (regexResult) {
+            const text = value.slice(caretposition < longest ? 0 : caretposition - longest, caretposition);
+            const aregexResult = symbolpatterns.exec(text);
+            const aaregexResult = apatterns.exec(text);
+            if (!aaregexResult && (!aregexResult || (caretposition <= longest ? regexResult.index < aregexResult.index : regexResult.index <= aregexResult.index))) {
+                insert = autocorrections[regexResult[0]] + (event.keyCode === 13 ? "\n" : insert);
+                deletecount = regexResult[0].length + 1;
+                output = true;
+            }
+        } else {
+            // Convert fractions and mathematical constants to Unicode characters
+            if (!output && fracts) {
+                // Numbers: https://regex101.com/r/7jUaSP/2
+                const re = /[0-9]+(\.[0-9]+)?$/;
+                const prevoustext = value.slice(0, caretposition - 1);
```

Comment:
Typo and please also use camelCase.

```suggestion
                const preivousText = value.slice(0, caretposition - 1);
```